app/controllers/cleaners_controller.rb

```diff
@@ -24,27 +26,42 @@ def edit
   # POST /cleaners
   # POST /cleaners.json
   def create
-    @cleaner = Cleaner.new(cleaner_params)
-
-    respond_to do |format|
+    city_id = params[:city_ids]
+    if !city_id.nil?
+      @cleaner = Cleaner.new(cleaner_params)
       if @cleaner.save
-        format.html { redirect_to @cleaner, notice: 'Cleaner was successfully created.' }
-        format.json { render :show, status: :created, location: @cleaner }
+        CitiesCleaner.add_city(@cleaner.id, params[:city_ids])
+        respond_to do |format|
+          format.html { redirect_to @cleaner, notice: 'Cleaner was successfully created.' }
+          format.json { render :show, status: :created, location: @cleaner }
+        end
       else
-        format.html { render :new }
-        format.json { render json: @cleaner.errors, status: :unprocessable_entity }
+        render new_cleaner_path
       end
+    else
+      flash[:notice] = 'Please select At least one city'
+      redirect_to new_cleaner_path
     end
   end
 
   # PATCH/PUT /cleaners/1
   # PATCH/PUT /cleaners/1.json
   def update
-    respond_to do |format|
+    id_city = params[:city_ids]
+    unless id_city.nil?
       if @cleaner.update(cleaner_params)
-        format.html { redirect_to @cleaner, notice: 'Cleaner was successfully updated.' }
-        format.json { render :show, status: :ok, location: @cleaner }
+        CitiesCleaner.where(cleaner_id: @cleaner.id).destroy_all
```

Comment:
@mayur-kambariya I assume this is not affecting existing bookings for this cleaner because it will generate new id for `CitiesCleaner` model1 @AnkurVyas-BTC